### PR TITLE
use block flag; revise docstring

### PR DESCRIPTION
**Pull request by lukeolson**

### CODE REVIEW

File: pyamg/strength.py [hunk 1]
```diff
@@ -121,25 +121,24 @@ def classical_strength_of_connection(A, theta=0.1, block=None, norm='abs'):
...
-        Threshold parameter in [0,1].
-    block : string, default None for CSR matrix and 'block' for BSR matrix
-        How to treat block structure of A:
-            None         : Compute SOC based on A as CSR matrix.
-            'block'      : Compute SOC based on norm of blocks of A.
-            'amalgamate' : Compute SOC based on A as CSR matrix, then compute
-                           norm of blocks in SOC matrix for a block SOC.
+        Threshold parameter in [0,1]
+    block : bool, default False
+        Compute strength of connection block-wise
```
Comment (jbschroder) [inline pyamg/strength.py:126]: dropping amalgate makes sense

**Review by jbschroder — APPROVED** [review]

File: pyamg/strength.py [hunk 2]
```diff
@@ -199,64 +202,41 @@ def classical_strength_of_connection(A, theta=0.1, block=None, norm='abs'):
...
+    S.eliminate_zeros()
 
-            # Take magnitude and scale by largest entry
-            S.data = np.abs(S.data)
-            S = scale_rows_by_largest_entry(S)
-            S.eliminate_zeros()
+    if blocksize > 1 and not block:
+        S = amalgamate(S, blocksize)
+        warn('Implicit conversion of A to bsr', sparse.SparseEfficiencyWarning)
+        A = A.tobsr(blocksize=[blocksize,blocksize])
```
Comment (bensworth) [inline pyamg/strength.py:239]: @lukeolson @jbschroder check if this makes sense to you. The only way blocksize is > 1 and block=False is we pass in a BSR matrix and block=False. If we then amalgamate the SOC matrix, we are mapping it to the block-dimensions of the original matrix. For this to work with other routines, e.g., interp/restrict, we need to map A back to a BSR matrix.

Comment (lukeolson) [reply pyamg/strength.py:239]: While `A` is converted within the routine, it's a copy (this is probably bad, but it's a different issue).  That last line `A = A.tobsr(blocksize=[blocksize,blocksize])` will create a new object that isn't returned.

So I think it's not needed? (I commented out that `tobsr()` line in the test below).  So I'd say go ahead.

```python
A = pyamg.gallery.poisson((4,)).tobsr(blocksize=(2,2))
print(f'{id(A)=}')

S = pyamg.strength.classical_strength_of_connection(A, block=True)
print(S.toarray())
print(A.toarray())
print(type(S))
print(type(A))

S = pyamg.strength.classical_strength_of_connection(A, block=False)
print(S.toarray())
print(A.toarray())
print(type(S))
print(type(A))
print(f'{id(A)=}')
```

output:
```
id(A)=4304610512
[[1.  0.5]
 [0.5 1. ]]
[[ 2. -1.  0.  0.]
 [-1.  2. -1.  0.]
 [ 0. -1.  2. -1.]
 [ 0.  0. -1.  2.]]
<class 'scipy.sparse._csr.csr_matrix'>
<class 'scipy.sparse._bsr.bsr_matrix'>

[[1. 1.]
 [1. 1.]]
[[ 2. -1.  0.  0.]
 [-1.  2. -1.  0.]
 [ 0. -1.  2. -1.]
 [ 0.  0. -1.  2.]]
<class 'scipy.sparse._csr.csr_matrix'>
<class 'scipy.sparse._bsr.bsr_matrix'>
id(A)=4304610512
```

Comment (jbschroder) [reply pyamg/strength.py]: Hi All...just read through the conversation today.  The revised code here to just do 

    if blocksize > 1 and not block:
        S = amalgamate(S, blocksize)

    return S

to end the function makes sense.

@Ben @lukeolson   Is this PR ready to merge now?

**Pull request merged**